### ISSUE TITLE
azure-native:dbforpostgresql/v20221201 does not allow serverName with hyphen

### ISSUE THREAD

**Issue by YuriiVarvynets** [pulumi/pulumi-azure-native#2186]
### What happened?

```
Diagnostics:
  azure-native:dbforpostgresql/v20221201:Server (test-test):
    error: azure-native:dbforpostgresql/v20221201:Server resource 'test-test' has a problem: 'serverName' does not match expression '^[a-z][a-z0-9]*$'
```

### Steps to reproduce

```
import pulumi_azure_native.dbforpostgresql.v20221201 as flexpostgre

flexible_server = flexpostgre.Server('test-test',
    server_name = 'test-test',
    ...
)
```

### Expected Behavior

`test-test` is valid name for PostgreSQL Server according to Azure Portal:
```
Server name must be at least 3 characters and at most 63 characters.
Server name must only contain lowercase letters, numbers, and hyphens. The server name must not start or end in a hyphen.
Server name must be available.
```

### Actual Behavior

`test-test` is not a valid name

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thomas11** [comment]
Hi @YuriiVarvynets, thank you for reporting this! I checked and unfortunately, the regex we use for validating this property comes directly from Microsoft, via the Azure OpenAPI spec. I've [reported it upstream](https://github.com/Azure/azure-rest-api-specs/issues/22302).

**Comment by YuriiVarvynets** [comment]
Thank you, @thomas11

**Comment by mikhailshilkov** [comment]
The upstream issue has been fixed, and we pulled those changes in the provider. @YuriiVarvynets could you please check if it's fixed with 1.94.0?

**Comment by YuriiVarvynets** [comment]
Thank you, @mikhailshilkov . I will check the fix as soon as I have time.
